python/src/puml/templates/requirements/requirements_diagram.puml.jinja2

```diff
@@ -1,21 +1,19 @@
-@startuml
+@startuml Requirements Diagram
 
 mainframe **req** [{{package | default('Package')}}] **{{title}}** [{{subtitle | default('Requirements')}}]
 
-{% for req in requirements %}
-object "<<{{req.type}}>>{% if req.title %}\n**{{req.title}}**{% endif %}" as {{req.name}} {
-    id = {{req.id}}
-    Text = {{req.shall}}
-    {% for attribute in req.attributes %}
+object "<<{{req_spec.type}}>>{% if req_spec.title %}\n**{{req_spec.title}}**{% endif %}" as {{req_spec.name}} {
+    id = {{req_spec.id}}
+    Text = {{req_spec.shall}}
+    {% for attribute in req_spec.attributes -%}
     {{attribute.name}} = {{attribute.value}}
     {% endfor %}
 }
 
-{% for r in req.connected %}
+{% for r in req_spec.connected -%}
 {{r.parent}} {{r.arrow | safe}} {{r.child}}{% if r.relationship %} : <<{{r.relationship}}>>{% endif %}
 
 {% endfor %}
```

Comment:
```suggestion
{% endfor -%}
```

Reply:
lot of extra spacing at the end of this again. Think the combo of adding the hyphen and removing some empty lines will help correcting that.

Reply:
Added the hyphen.